db/migrate/20170124141701_add_uniq_index_on_minions_id.rb

```diff
@@ -0,0 +1,11 @@
+class AddUniqIndexOnMinionsId < ActiveRecord::Migration[5.0]
+  def up
+    remove_index :minions, :hostname
+    add_index :minions, :hostname, unique: true
+  end
+
+  def down
+    remove_index :minions, :hostname
+    add_index :minions, :hostname
+  end
+end
```

Comment:
TBH I don't really like the idea of removing an index for adding it afterwards (the table will be empty, yeah), but maybe you can just modify the original migration at this point.